classes/core/traits/HasForeignKey.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @file classes/core/traits/HasForeignKey.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class HasForeignKey
+ *
+ * @brief A trait to determine the existence foreign key in a table
+ *
+ */
+
+namespace PKP\core\traits;
+
+use Illuminate\Support\Facades\Schema;
+
+trait HasForeignKey
```

Comment:
I'm not sure a trait is the right place for this. It implies that the class that uses the trait has a foreign key, when actually this adds a way to check whether a database does. And in general I'm hesitant to add support code that's only called from the migrations, since migrations are throw-away (meaning they will eventually be removed when the upgrade is deemed too old to support), and when that happens we'll be left with unused traits lying around. (This is particularly a risk with traits, since they don't need any particular inheritance heirarchy and will pile up in the general purpose `traits` directory.)

I can think of two approaches:
 - Extend the Schema facade to add a general purpose `hasForeignKey` function, then call it there. If we needed them, an extended Schema facade would also be a more sensible home for some of the `Installer` functions that you've also adapted from Doctrine DBAL; they aren't currently used in `main` but have in the past (and may be again in the future) in the `upgrade.xml` script. See e.g. the `condition` attribute in this upgrade line for 3.3.0-x: https://github.com/pkp/ojs/blob/stable-3_3_0/dbscripts/xml/upgrade.xml#L24C1-L24C11. However, see my comments on those later, though -- it looks like those can just be removed entirely.
 *...or...*
 - Just continue ot use `Schema::getForeignKeys` directly, even though it'll mean a little bit of duplicated `->pluck(...)->contains(...)` code in the migrations. It's not much.

Reply:
@asmecher extending the schema facade just for this seems a bit overkill to me . That will make to register it and maintaining it. I am wondering if this is something we can put in our core migration , that is `PKP\migration\Migration` . One file and we can remove it later if we need to . Placing following code 
```php
collect(Schema::getForeignKeys($tableName))
            ->pluck('name')
            ->contains($keyName)
```
in 10/15 places also seems too much duplication . What you say ?